thorlcr/activities/join/thjoinslave.cpp

```diff
@@ -371,38 +368,43 @@ class JoinSlaveActivity : public CSlaveActivity, implements ILookAheadStopNotify
         else
             stopRightInput();
     }
-    virtual void abort()
+    virtual void abort() override
     {
         CSlaveActivity::abort();
         if (joinhelper)
             joinhelper->stop();
     }
-    virtual void stop()
+    virtual void stop() override
     {
         stopLeftInput();
         stopRightInput();
-        lhsProgressCount = joinhelper->getLhsProgress();
-        rhsProgressCount = joinhelper->getRhsProgress();
-        {
-            CriticalBlock b(joinHelperCrit);
-            joinhelper.clear();
-        }
-        ActPrintLog("SortJoinSlaveActivity::stop");
-        rightStream.clear();
-        if (!islocal) {
-            unsigned bn=noSortPartitionSide()?2:4;
-            ActPrintLog("JOIN waiting barrier.%d",bn);
-            barrier->wait(false);
-            ActPrintLog("JOIN barrier.%d raised",bn);
-            sorter->stopMerge();
+        if (queryInputStarted(0))
```

Comment:
I need to chain stops, as roxie does, even if this act hasn't been started.
This is related to the previous comment, re. stopping conditional inputs asap so that upstream splitters know asap and can avoid buffering.
But need to be careful to avoid anything in the activity stop code that hasn't been initialized if never started , as in this case.